content/tutorials/react-admin-panel-with-ably-and-refine.textile

```diff
@@ -0,0 +1,289 @@
+
+```[nodejs]
+import { Ably } from "@refinedev/ably";
+
+export const ablyClient = new Ably.Realtime(process.env.REACT_APP_ABLY_API_KEY!);
+```
+
+Similarly, copy and paste the code below into the @src/utility/index.ts@ file you created. It will enable you to import exports from the @src/utility@ directory.
+
+```[nodejs]
+export * from "./ablyClient";
+```
+
+The @<Refine />@ component is the entry point for any refine application. The App component renders the @<Refine />@ component in the @App.tsx@ file. The @liveProvider@ prop of the @<Refine />@ component is for integrating refine with any realtime provider.
+
+The @liveProvider@ prop is agnostic of any realtime solution. You can use it with any realtime provider. In this article, you will use it to integrate Ably with your refine application. Apply the changes below to your @App.tsx@ file.
+
+```[nodejs]
+...
+//highlight-start
+import { ablyClient } from "utility";
+import { liveProvider } from "@refinedev/ably";
+//highlight-end
+
+function App() {
+  return (
+          ...
+            <Refine
+              ...
+              //highlight-next-line
+              liveProvider={liveProvider(ablyClient)}
+              options={{
+              ...
+                //highlight-next-line
+                liveMode: "auto",
+              }}
+            >
+            ...
+  );
+}
+
+export default App;
+```
+
+h3. liveMode
+
+Quick info for live mode options.
+
+@auto@: Queries of related resource are invalidated in Realtime as new events from subscription arrive.
+
+@manual@: Queries of related resource are not invalidated in Realtime, instead "onLiveEvent":https://refine.dev/docs/api-reference/core/providers/live-provider/#onliveevent is run with the event as new events from subscription arrive.
+
+@off@: Disables live mode.
+
+"Refer to live provider docs for more information":https://refine.dev/docs/api-reference/core/providers/live-provider/
+
+h3. We are ready for testion realtime feature on the app
+
+That is everything you need to do to integrate Ably with your refine application.
+
+You can now perform CRUD operations on the @blog_posts@ in the fake API we used. When you delete, update, or create a new product, Ably will broadcast the changes in realtime.
+
+As demonstrated below, any modifications made to an existing record are immediately reflected, all thanks to the Ably live provider.
+
+<a href="/images/tutorials/ably-refine/img4.gif" target="_blank">
+  <img src="/images/tutorials/ably-refine/img4.gif" style="width: 100%"
+  title="Click to enlarge" alt="The realtime dashboard working">
+</a>
+
+This tutorial illustrates how you can build a basic CRUD app with realtime feature with refine and Ably. There is a lot you can do with refine and Ably that I have not highlighted in this article. Check out the "refine":https://refine.dev/docs/ or "Ably":https://www.ably.com/docs documentation.
```

Comment:
"...that I have not..." 

I think need to reword this from the 1st person if we're leaving the author blank.